cep-20.1.md

```diff
@@ -0,0 +1,286 @@
+# A new recipe format – part 1
+
+<table>
+<tr><td> Title </td><td> A new recipe format </td>
+<tr><td> Status </td><td> Proposed</td></tr>
+<tr><td> Author(s) </td><td> Wolf Vollprecht &lt;wolf@prefix.dev&gt;</td></tr>
+<tr><td> Created </td><td> May 23, 2023</td></tr>
+<tr><td> Updated </td><td> May 23, 2023</td></tr>
+<tr><td> Discussion </td><td>  </td></tr>
+<tr><td> Implementation </td><td>https://github.com/prefix-dev/rattler-build</td></tr>
+</table>
+
+## Abstract
+
+We propose a new recipe format that is heavily inspired by conda-build. The main
+change is a pure YAML format without arbitrary Jinja or comments with semantic
+meaning.
+
+## Motivation
+
+The conda-build format has grown over the years to become quite complex.
+Unfortunately it has never been formally "specified" and it has grown some
+features over time that make it hard to parse as straightforward YAML.
+
+The CEP attempts to introduce a subset of the conda build format that allows for
+fast parsing and building of recipes.
+
+### History
+
+A discussion was started on what a new recipe spec could or should look like.
+The fragments of this discussion can be found here:
+https://github.com/mamba-org/conda-specs/blob/master/proposed_specs/recipe.md
+The reason for a new spec are:
+
+- Make it easier to parse ("pure yaml"). conda-build uses a mix of comments and
+  jinja to achieve a great deal of flexibility, but it's hard to parse the
+  recipe with a computer
+- iron out some inconsistencies around multiple outputs (build vs. build/script
+  and more)
+- remove any need for recursive parsing & solving
+
+## Major differences with conda-build
+
+- no full Jinja2 support: no conditional or `{% set ...` support, only string
+  interpolation. Variables can be set in the toplevel "context" which is valid
+  YAML
+- Jinja string interpolation needs to be quoted at the beginning of a string,
+  e.g. `- "{{ version }}"` in order for it to be valid YAML
+- Selectors use a YAML dictionary style (vs. comments in conda-build). E.g. `-
+  sel(osx): somepkg` instead of `- somepkg  # [osx]`
+
+## Selectors
+
+Selectors in the new spec take the following format:
+
+`sel(unix): selected_value`
```

Comment:
How about a simplified variant:
`(unix): selected_value`
`(win and arm64)`
`((osx or linux) and aarch64)
`(something == "test")

Reply:
Looks like this is also valid YAML:

<img width="1022" alt="Screenshot 2023-06-07 at 14 49 00" src="https://github.com/conda-incubator/ceps/assets/885054/9a84ee1a-ff50-463d-8759-a280b878c4ef">

However, it would effectively special case all dictionary keys that start with a round bracket. I think that would be OK, but less explicit than the `sel(...)` operator.

Reply:
Not to bikeshed, but what about something like `is(unix)`?